exonum-java-binding/core/rust/src/storage/db.rs

```diff
@@ -28,104 +28,120 @@ pub(crate) type Value = Vec<u8>;
 ///    in the case when the java side creates and owns it;
 /// - it just holds a reference, when one is provided from the rust side.
 ///
-/// For storage API we need a reference, so we create it to the owned part. But since there is no
-/// way in Rust to make a `View` value not movable. Furthermore, it have to be moved from the stack
-/// to the heap in order to be converted into `Handle` for the java side. So a `Fork` value
-/// should be placed in the heap to prevent its movement after creating a reference to it.
-
-pub(crate) struct View {
-    /// Unsafe reference to the owned View (owned either by Rust or by Java)
-    reference: ViewRef,
-    /// Allows to drop owned view if needed
-    owned: Option<ViewOwned>,
+/// As `View` does not have a lifetime, nothing protects us from dereferencing a freed reference,
+/// the extreme caution must be taken in places where `View` is constructed and send to Java.
+/// Java code must never store a `View` handle beyond the scope it was initially acquired
```

Comment:
Not sure I understand what you mean here. Usually, Java releases a handle, and it must not be used subsequently. Is it anyhow different for this thing?

Reply:
Yep, no difference here. Java can't store handle to `View::Ref*` variants, because noone can verify that the reference will be still valid. Do you think I should remove this sentence as `NativeHandle` docs covers this enough? I just wasn't able to find any docs in native code explaining this mechanics.

Reply:
> Java can't store handle to View::Ref* variants, because noone can verify that the reference will be still valid.

Reply:
What do you mean by Java not being able to store a handle to ref variants? How could it use them then? Or do you mean for longer than the method invocation?

Reply:
Yep, for longer than the method invocation (that was the purpose of "beyond the scope it was initially acquired") 🤔 Probably I should avoid such phrases, my fault.